### PR TITLE
feat: added a crude progress bar and a please wait message

### PR DESCRIPTION
**Pull request by drorrose**

### CODE REVIEW

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
The progress won't always work like that..

**Comment by drorrose** [comment]
@davidmeirlevy 
Reimplemented version is ready for review!

File: controllers/start.js [hunk 1]
```diff
@@ -30,20 +31,27 @@ async function startCommand (mode = 'user', options) {
...
-	execSync( ['npm', ...appArgs].join(' '), childArgs);
+	console.log(blue('Initializing Greenpress..\n'));
+	console.log(blue('Doing our magic, might take a few minutes. Please wait.\n'));
+
+	const child = spawn('npm', appArgs, childArgs);
+
+	child.on('error', (err) => {
+		console.log(red(`\nAn error occured while starting greenpress! Error:\n`), err);
+		process.exit(1);
+	});
```
Comment (davidmeirlevy) [inline controllers/start.js:42]: I would suggest to take that code that initialize greenpress, and put it in a separate function.
That function will build the args, run the spawn, subscribe to error, and return the process instance.

File: controllers/start.js [hunk 2]
```diff
@@ -30,20 +31,27 @@ async function startCommand (mode = 'user', options) {
...
+		console.log(`\rTo enter your app: http://localhost:3000`);
 		process.exit(0);
-	} 
-
-	console.log('Server took to long to run');
-	process.exit(1);
+	} catch (err) {
+		console.log('An error occured during server startup');
+		process.exit(1);
+	}
```
Comment (davidmeirlevy) [inline controllers/start.js:54]: This all area can be wrapped in a function.

File: services/start.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
-const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+const { spawn } = require('child_process');
+
+const imagesProgressBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+const maxImagesBarValue = 20;
+let imagesCurrentValue = 0;
+
+const servicesProgressBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
+const maxServicesBarValue = 80;
+let servicesCurrentValue = 0;
```
Comment (davidmeirlevy) [inline services/start.js:14]: This file is too long, handles various of things and couples to different behaviors.
Separate the state and the progressBar management into different files.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: services/start.js (hunk 3 above)
Comment (drorrose) [reply services/start.js]: Done, created a separate utility file for it

File: controllers/start.js (hunk 2 above)
Comment (drorrose) [reply controllers/start.js]: Done

File: controllers/start.js (hunk 1 above)
Comment (drorrose) [reply controllers/start.js]: Done

**Review by or-yam — APPROVED** [review]

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**